File: testrust/tphi.rs
const LIMIT: u64 = 40;

fn main()
{
    println!( "phi should tend towards 1.61803398874989484820458683436563811772030" );
    let mut prev2: u128 = 1;
    let mut prev1: u128 = 1;
    let mut last_shown: u64 = 0;
    let mut _i: u64;

    for _i in 1..LIMIT+1 {
        let next: u128 = prev1 + prev2;
        prev2 = prev1;
        prev1 = next;

        if _i == ( last_shown + 5 )
        {
            last_shown = _i;
            println!( "  at {:4} iterations: {}", _i, prev1 as f64 / prev2 as f64 );
        }
    }

    println!( "exiting finding phi with great success" );
}

File: c_tests/tphi.c
#include <stdio.h>
#include <stdint.h>

int main()
{
    printf( "should tend towards 1.61803398874989484820458683436563811772030\n" );
    const uint64_t limit = 40;
    unsigned __int128 prev2 = 1;
    unsigned __int128 prev1 = 1;
    uint64_t last_shown = 0;

    for ( uint64_t i = 1; i <= limit; i++ )
    {
        unsigned __int128 next = prev1 + prev2;
        prev2 = prev1;
        prev1 = next;

        if ( i == ( last_shown + 5 ) )
        {
            last_shown = i;
            printf( "  at %2u iterations: %.16lf\n", (unsigned int) i, (double) prev1 / (double) prev2 );
        }
    }

    printf( "done\n" );
    return 0;
}
